File: DSA/Data_Structure/Binary_Tree/Easy/Binary_Tree_Inorder_Traversal/Solutions/Solve.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
 var inorderTraversal = function(root) {
    const stack = [];
    const res = []
    if(!root) return res;
    let node = root;
    while (node || stack.length) {
        while(node){
            stack.push(node);
            node = node.left;
        }
        
        node = stack.pop();
        res.push(node.val);
        
        node = node.right;
    }
    return res;
};

File: DSA/Algorithm/Two_Pointers/Easy/Squares_of_a_Sorted_Array/Solutions/solve.py
def sortedSquares(nums):
    left = 0
    right = len(nums) - 1
    res = [None]*len(nums)
    i = len(nums) - 1
    while(i >= 0):
        if nums[left] ** 2 > nums[right] ** 2:
            res[i] = nums[left] ** 2
            left += 1
            i -= 1
        else:
            res[i] = nums[right] ** 2
            right -= 1
            i -= 1
    return res

def main():
    print(sortedSquares([-7,-3,2,3,11]))

main()